x/auth/ante.go

```diff
@@ -95,8 +96,15 @@ func NewAnteHandler(am AccountMapper, fck FeeCollectionKeeper) sdk.AnteHandler {
 			}
 
 			// first sig pays the fees
-			// TODO: Add min fees
 			// Can this function be moved outside of the loop?
+			if ctx.IsCheckTx() && !simulate {
+				// required fees must be greater than the minimum set by the validator
+				if ctx.MinimumFees().Minus(fee.Amount).IsNotNegative() {
+					fee = NewStdFee(fee.Gas, ctx.MinimumFees()...)
```

Comment:
Why are we setting `fee` here? Do we only want users to pay the minimum fee even if they set a higher one?

Reply:
When `MinimumFees > Fee` (`MinimumFees - Fee > 0`), we pick `MinimumFees`

Reply:
What if the user supplies `Fee > MinimumFees`?

Reply:
We shouldn't auto set the fee to min fee if the provided fee isn't enough. There are several issues with that. It will cause a consensus fault, since every validator would subtract it differently. Even if this didn't happen, it provides a validator a way to drain your entire account.

Reply:
We should do neither - if the user didn't sign a transaction with high-enough fees, we should reject the transaction, not subtract fees from their account which they didn't sign. The only check we should perform is that the fees provided in the transaction are greater than the minimum fee - the user should always pay exactly the fee they specified.

Reply:
Fixed. Please review @cwgoes @ValarDragon @alexanderbez